controller/api/destination/external-workload/endpoints_controller.go

```diff
@@ -0,0 +1,513 @@
+					ec.enqueueUpdate(svc)
+				}
+				return
+			}
+
+			tomb, ok := obj.(cache.DeletedFinalStateUnknown)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: couldn't retrieve obj from DeletedFinalStateUnknown %#v", obj)
+				return
+			}
+
+			ew, ok = tomb.Obj.(*ewv1alpha1.ExternalWorkload)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: DeletedFinalStateUnknown contained object that is not a v1alpha1.ExternalWorkload %#v", obj)
+				return
+			}
+
+			services, err := ec.getSvcMembership(ew)
+			if err != nil {
+				ec.log.Errorf("failed to get service membership for %s/%s: %v", ew.Namespace, ew.Name, err)
+				return
+			}
+
+			for svc := range services {
+				ec.enqueueUpdate(svc)
+			}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			old, ok := oldObj.(*ewv1alpha1.ExternalWorkload)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: expected *v1alpha1.ExternalWorkload, got %#v", oldObj)
+			}
+
+			updated, ok := newObj.(*ewv1alpha1.ExternalWorkload)
+			if !ok {
+				ec.log.Errorf("error processing ExternalWorkload event: expected *v1alpha1.ExternalWorkload, got %#v", newObj)
+			}
+
+			// Ignore resync updates. If nothing has changed in the object, then
+			// the update processing is redudant.
+			if old.ResourceVersion == updated.ResourceVersion {
+				ec.log.Tracef("skipping ExternalWorkload resync update event")
+				return
+			}
+
+			// Compute whether any labels have changed in between updates. If
+			// they have, this might affect service membership so we need to
+			// process the update.
+			labelsChanged := false
+			if len(old.Labels) != len(updated.Labels) {
+				labelsChanged = true
+			} else {
+				for upK, upV := range updated.Labels {
+					oldV, ok := old.Labels[upK]
+					if ok && oldV == upV {
+						// Labels match
+						continue
+					} else {
+						labelsChanged = true
+						break
+					}
+				}
+			}
+
+			readinessChanged := isReady(old) != isReady(updated)
+
+			// Check if anything in the spec has changed in between versions.
+			specChanged := false
+			ports := make(map[int32]struct{})
+			for _, pSpec := range updated.Spec.Ports {
+				ports[pSpec.Port] = struct{}{}
+			}
+			for _, pSpec := range old.Spec.Ports {
+				if _, ok := ports[pSpec.Port]; !ok {
+					specChanged = true
+					break
+				}
+			}
+
+			ips := make(map[string]struct{})
+			for _, addr := range updated.Spec.WorkloadIPs {
+				ips[addr.Ip] = struct{}{}
+			}
+			for _, addr := range old.Spec.WorkloadIPs {
+				if _, ok := ips[addr.Ip]; !ok {
+					specChanged = true
+					break
+				}
+			}
+
+			// If nothing has changed, don't bother with the update
+			if !readinessChanged && !specChanged && !labelsChanged {
+				ec.log.Debugf("skipping ExternalWorkload update; nothing has changed between old rv %s and new rv %s", old.ResourceVersion, updated.ResourceVersion)
+				return
+			}
+
+			services, err := ec.getSvcMembership(updated)
```

Comment:
Do we also have to recompute membership in services that old is part of?  In case this workload is no longer part of a service?

Reply:
Yup! Labels changing in between updates is an indication that the membership has to be recomputed for the old service too.